gostackparse.go

```diff
@@ -69,17 +70,30 @@ func Parse(r io.Reader) ([]*Goroutine, []error) {
 
 	statemachine:
 		switch state {
-		case stateHeader:
+		case stateHeader, stateOriginatingFrom:
 			// Ignore lines that don't look like goroutine headers. This helps with
 			// leading/trailing whitespace but also in case we encountered an error
 			// during the previous goroutine and need to seek to the beginning of the
 			// next one.
-			if !bytes.HasPrefix(line, goroutinePrefix) {
-				continue
+			if state == stateHeader {
+				if !bytes.HasPrefix(line, goroutinePrefix) {
+					continue
+				}
+				g = parseGoroutineHeader(line[len(goroutinePrefix):])
+				goroutines = append(goroutines, g)
+			}
+			if state == stateOriginatingFrom {
```

Comment:
NIT: This should probably be an `else if` for the above?

Reply:
Yep whoops. Good ol branch prediction will optimize that away